core/src/main/java/org/elasticsearch/transport/local/LocalTransport.java

```diff
@@ -288,6 +294,9 @@ private void handleRequest(StreamInput stream, long requestId, LocalTransport so
             if (reg == null) {
                 throw new ActionNotFoundTransportException("Action [" + action + "] not found");
             }
+            // reserve here in order be able to send a response in case we break the limit
+            inFlightRequestsBreaker.addEstimateBytesAndMaybeBreak(messageLengthBytes, "Node in-flight requests");
+            transportChannel.setReservedBytes(messageLengthBytes);
```

Comment:
can we make this part of the constructor of the transportChannel? this also means we need to change the breaker earlier (which is fine because this is called under a try catch anyway.